src/compiler/commandLineParser.ts

```diff
@@ -371,13 +371,25 @@ namespace ts {
                     parseResponseFile(s.slice(1));
                 }
                 else if (s.charCodeAt(0) === CharacterCodes.minus) {
-                    s = s.slice(s.charCodeAt(1) === CharacterCodes.minus ? 2 : 1).toLowerCase();
+                    s = s.slice(s.charCodeAt(1) === CharacterCodes.minus ? 2 : 1);
 
                     // Try to translate short option names to their full equivalents.
-                    if (hasProperty(shortOptionNames, s)) {
+                    if (hasProperty(shortOptionNames, s.toLowerCase())) {
                         s = shortOptionNames[s];
                     }
+                    else {

```

Comment:
It's weird that this doesn't work for `-t=es3` because you try to do the shorthand lookup first.